platform/lib/pipeline/componentReferenceImporter.js

```diff
@@ -72,25 +74,42 @@ class ComponentReferenceImporter {
     // Keep track of all promises to complete function
     const savedDocuments = [];
 
+    const versionMapping = {};
     for (const extension of extensions) {
       const documents = await this._findExtensionDocs(extension);
       const versions = [...new Set(documents.map((doc) => doc.version).sort().reverse())];
 
+
       if (!documents.length) {
         log.warn(`No matching document for component: ${extension.name}`);
       } else {
+        const supportedVersions = {};
+        versionMapping[extension.name] = supportedVersions;
         documents.forEach((doc) => {
           // IMPORTANT: first rewrite URLs
           this._rewriteRelativePaths(extension.path, doc.document);
           // then set versions to avoid re-writing the relative version toggle links
           this._setMetadata(
               doc.tagName || extension.name, doc.document, doc.version, versions);
+          if (doc.document.isCurrent) {
+            supportedVersions.current = doc.version;
+          }
+          doc.document.stripInlineTitle();
           savedDocuments.push(
               this._saveDocument(doc.tagName || extension.name, doc.document, doc.version));
+          doc.document.formats.forEach((format) => {
+            let versionsPerFormat = supportedVersions[format];
+            if (!versionsPerFormat) {
+              versionsPerFormat = [];
+              supportedVersions[format] = versionsPerFormat;
+            }
+            versionsPerFormat.push(doc.version);
+          });
         });
       }
     }
 
+    writeFileSync(FORMAT_COMPONENT_MAPPING, JSON.stringify(versionMapping, null, 2), 'utf-8');
```

Comment:
nit: most other stuff in here is async, and you're returning promises anyway. No big deal though.

Reply:
done